platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/ValueSanitizer.java

```diff
@@ -20,5 +20,5 @@
 
 public interface ValueSanitizer<T> {
     @Nullable
-    T sanitize(@Nullable T value);
+    T sanitize(Class<?> valueType, @Nullable T value);
```

Comment:
❔ Do you think that storing the type in the sanitizer instance (and, by extension, having a per-property instance when enum is involved) is going to take too much memory? Otherwise we could have this interface intact.

Reply:
I think it should not be a big problem. I can also implement per-property instance

Reply:
Done